engine/src/main/java/io/zeebe/engine/processing/bpmn/container/CallActivityProcessor.java

```diff
@@ -165,21 +138,35 @@ public void beforeExecutionPathCompleted(
   @Override
   public void afterExecutionPathCompleted(
       final ExecutableCallActivity element,
-      final BpmnElementContext flowScopeContext,
-      final BpmnElementContext childContext) {}
+      final BpmnElementContext callActivityContext,
+      final BpmnElementContext childContext) {
+    stateTransitionBehavior.completeElement(callActivityContext);
+  }
 
   @Override
   public void onChildTerminated(
       final ExecutableCallActivity element,
       final BpmnElementContext callActivityContext,
       final BpmnElementContext childContext) {
     final var currentState = callActivityContext.getIntent();
-    if (currentState == ProcessInstanceIntent.ELEMENT_TERMINATING) {
-      stateTransitionBehavior.transitionToTerminated(callActivityContext);
-    } else {
+    if (currentState != ProcessInstanceIntent.ELEMENT_TERMINATING) {
       final var message = String.format(UNABLE_TO_TERMINATE_FROM_STATE_MESSAGE, currentState);
       throw new BpmnProcessingException(callActivityContext, message);
     }
+
+    eventSubscriptionBehavior
+        .findEventTrigger(callActivityContext)
+        .ifPresentOrElse(
+            eventTrigger -> {
+              final var terminated =
+                  stateTransitionBehavior.transitionToTerminated(callActivityContext);
+              eventSubscriptionBehavior.activateTriggeredEvent(terminated, eventTrigger);
+            },
+            () -> {
+              final var terminated =
+                  stateTransitionBehavior.transitionToTerminated(callActivityContext);
+              stateTransitionBehavior.onElementTerminated(element, terminated);
+            });
```

Comment:
This is common behaviour for all containers isn't it? More of a nit, but I guess we could think about how to handle this in the future in a common way for all containers.